### PR TITLE
TRON-133 use bitcoin key generation libs

### PR DESCRIPTION
**Pull request by argjv**
[TRON-133](https://bitgoinc.atlassian.net/browse/TRON-133) Use bitcoin account generation methods instead of Tron Web ones.

### CODE REVIEW

File: test/unit/coin/trx/util.ts [hunk 1]
```diff
@@ -16,7 +16,16 @@ describe('Util library should', function() {
   const pub = '046EBFB90C396B4A3B992B727CB4714A32E2A6DE43FDB3EC266286AC2246D8FD1E23E12C0DEB752C631A9011BBF8B56E2FBAA20E99D3952F0A558D11F96E7C1C5D';
   const addressHex = '412C2BA4A9FF6C53207DC5B686BFECF75EA7B80577';
   const base58 = 'TDzm1tCXM2YS1PDa3GoXSvxdy4AgwVbBPE';
-  const addrBytes = [ 65, 44, 43, 164, 169, 255, 108, 83, 32, 125, 197, 182,134,191,236,247,94,167,184,5,119 ]
+  const addrBytes = [ 65, 44, 43, 164, 169, 255, 108, 83, 32, 125, 197, 182,134,191,236,247,94,167,184,5,119 ];
+
+  it('generate a valid account', () => {
+    const seed = Buffer.alloc(32);
```
Comment (twtaylorbitgo) [inline test/unit/coin/trx/util.ts]: we should pass a known seed to this in the test

Comment (argjv) [reply test/unit/coin/trx/util.ts]: It is a known seed, it's all zeros

Comment (twtaylorbitgo) [reply test/unit/coin/trx/util.ts]: eh fair enough, I think we should be a little more explicit but that works

File: test/unit/coin/trx/util.ts [hunk 2]
```diff
@@ -16,7 +16,16 @@ describe('Util library should', function() {
   const pub = '046EBFB90C396B4A3B992B727CB4714A32E2A6DE43FDB3EC266286AC2246D8FD1E23E12C0DEB752C631A9011BBF8B56E2FBAA20E99D3952F0A558D11F96E7C1C5D';
   const addressHex = '412C2BA4A9FF6C53207DC5B686BFECF75EA7B80577';
   const base58 = 'TDzm1tCXM2YS1PDa3GoXSvxdy4AgwVbBPE';
-  const addrBytes = [ 65, 44, 43, 164, 169, 255, 108, 83, 32, 125, 197, 182,134,191,236,247,94,167,184,5,119 ]
+  const addrBytes = [ 65, 44, 43, 164, 169, 255, 108, 83, 32, 125, 197, 182,134,191,236,247,94,167,184,5,119 ];
+
+  it('generate a valid account', () => {
```
Comment (twtaylorbitgo) [inline test/unit/coin/trx/util.ts]: we should have an additional test where we pass no see (to make sure the crypto.randomBytes path spits out a valid key

File: src/coin/trx/utils.ts [hunk 3]
```diff
@@ -20,8 +22,66 @@ import { UtilsError } from '../baseCoin/errors';
...
+    seed = crypto.randomBytes(512 / 8);
+  }
+
+  const extendedKey = HDNode.fromSeedBuffer(seed);
+
+  // convert our prv, pub
+  const prv = extendedKey.keyPair.getPrivateKeyBuffer().toString('hex').toUpperCase();
+  const pub = extendedKey.keyPair.getPublicKeyBuffer().toString('hex').toUpperCase();
+
+  validateKeyPair(pub, prv);
```
Comment (tylerlevine) [inline src/coin/trx/utils.ts]: Why do we need to validate here? Does HDNode give you invalid keys with some nonzero probability?

File: src/coin/trx/utils.ts [hunk 4]
```diff
@@ -20,8 +22,66 @@ import { UtilsError } from '../baseCoin/errors';
...
+    publicKey,
+    address: {
+      base58: getBase58AddressFromByteArray(addressBytes),
+      hex: getHexAddressFromByteArray(addressBytes),
+    },
+  };
+}
+
+/**
+ * Validate the public and private key are valid points of the secp256k1 elliptic curve.
```
Comment (tylerlevine) [inline src/coin/trx/utils.ts]: Nit: The private key is a scalar (number), only the public key is a point on secp.

File: src/coin/trx/utils.ts [hunk 5]
```diff
@@ -20,8 +22,66 @@ import { UtilsError } from '../baseCoin/errors';
...
+
+  // convert our prv, pub
+  const prv = extendedKey.keyPair.getPrivateKeyBuffer().toString('hex').toUpperCase();
+  const pub = extendedKey.keyPair.getPublicKeyBuffer().toString('hex').toUpperCase();
+
+  validateKeyPair(pub, prv);
+
+  // 0 pads the private key
+  let priKeyHex = prv;
+  while (priKeyHex.length < 64) {
```
Comment (tylerlevine) [inline src/coin/trx/utils.ts]: This has a bit of code smell to it as well, but I don't think it's worth bringing in lodash for one util function.

Comment (twtaylorbitgo) [reply src/coin/trx/utils.ts]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Comment (twtaylorbitgo) [reply src/coin/trx/utils.ts]: we'd have to upgrade the webpack targets though

File: src/coin/trx/utils.ts (hunk 3 above)
Comment (argjv) [reply src/coin/trx/utils.ts]: Not really, but this was added in the beginning when we implemented our key generation logic. We wanted to triple check we did it right.

File: src/coin/trx/utils.ts (hunk 5 above)
Comment (argjv) [reply src/coin/trx/utils.ts]: Fixed by using KeyPair objects from bitgo-utxo-lib which automatically pads the prv.

File: src/coin/trx/utils.ts (hunk 3 above)
Comment (tylerlevine) [reply src/coin/trx/utils.ts]: This is better off as a unit test then - we shouldn't be testing our deps in the hot path

**Pull request merged**